CONTRIBUTING.md

```diff
@@ -0,0 +1,297 @@
+(such that "2 positive reviews from maintainers" requirement will remain 
+achievable).
+
+
+## Coding conventions
+
+Overall, this library must reflect Bitcoin Core approach whenever possible.
+However, since many of the things in Bitcoin Core are maintained due to 
+historical reasons and may represent poor design, Rust-idiomatic style is 
+preferred to "how it looks in Core" if everyone agrees.
+
+### Formatting
+
+We plan to utilize `rustfmt` for keeping the code formatting consistent. 
+However, this will be a gradual process since a thorough peer review is required
+to make sure that no unintended changes are introduced with the change of 
+formatting. Thus, all PRs introducing large blocks of re-formatted code will not
+be reviewed.
+
+The current plan is to phase it in over multiple commits or even multiple PRs,
+which will introduce no other changes that re-formatting, such that each change 
+may be independently re-reproduced by each reviewer. The first commit should add
+attributes to disable fmt for some parts of the code and a second one does the 
+formatting – so only the first one needs review, the rest will be reproducible.
+
+You may check the [discussion on the formatting](https://github.com/rust-bitcoin/rust-bitcoin/issues/172)
+and [how it is planned to coordinate it with crate refactoring](https://github.com/rust-bitcoin/rust-bitcoin/pull/525)
+
+<!--
+Rust-fmt should be used as a coding style recommendations in general, with a
+default coding style. By default, Rustfmt uses a style which conforms to the
+[Rust style guide][style guide] that has been formalized through the [style RFC
+process][fmt rfcs]. It is also required to run `cargo fmt` to make the code
+formatted according to `rustfmt` parameters 
+-->
+
+### Derivation
+
+Derivations applied to a data structures should be standardized:
+
+1. All non-error types should opportunistically derive, where it is possible,
+   the following traits:
+   - `Copy`
+   - `Clone`
+   - `PartialEq` and `Eq`
+   - `PartialOrd` and `Ord`
+   - `Hash`
+   - `Debug`
+   
+   By "where possible" we mean that by default a code line
+   ```rust
+   #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+   ```
+   must be placed before each struct, and then those of these traits, which can't
+   be auto-derived because of the member field restrictions should be removed.
+
+2. `Eq`, `PartialEq`, `Ord`, `PartialOrd` derivation must be skipped/removed
+   from pt. 1 in the following situations:
+   - for types that don't have reflexive equality/ordering
+   - types which has lexicographic ordering defined as a part of a standard must
+     provide manual derivation
+   - types which may be more efficiently compared with bitcoin-specific rules
+     should provide manual derivation
+
+3. `Debug` must not be derived on structs and enums which may contain secret
+   data, and a manual `Debug` implementation should be provided instead.
```

Comment:
I think "... implementation excluding private data should be ..." would be clearer